// react-stormpath/lib/stores/UserStore.js
'use strict';

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _app = require('../app');

var _app2 = _interopRequireDefault(_app);

var BaseStore = require('../stores/BaseStore');
var UserService = require('../services/UserService');
var UserConstants = require('../constants/UserConstants');

var _session = false;
var _sessionError = null;
var _sessionResolved = false;

var UserStore = (function (_BaseStore) {
  _inherits(UserStore, _BaseStore);

  function UserStore() {
    _classCallCheck(this, UserStore);

    _BaseStore.call(this);
    this.service = null;
  }

  UserStore.prototype.init = function init(context) {
    this.service = new UserService(context.getEndpoints());
    this.resolveSession();
  };

  UserStore.prototype.getSession = function getSession() {
    return _session;
  };

  UserStore.prototype.isAuthenticated = function isAuthenticated(callback) {
    this.resolveSession(function (err, result) {
      callback(err, !err && _session !== false);
    });
  };

  UserStore.prototype.login = function login(options, callback) {
    var self = this;

    this.reset();

    this.service.login(options, function (err, result) {
      if (err) {
        return callback(err);
      }

      self.resolveSession(callback);
    });
  };

  UserStore.prototype.register = function register(options, callback) {
    this.service.register(options, callback);
  };

  UserStore.prototype.forgotPassword = function forgotPassword(options, callback) {
    this.service.forgotPassword(options, callback);
  };

  UserStore.prototype.changePassword = function changePassword(options, callback) {
    this.service.changePassword(options, callback);
  };

  UserStore.prototype.verifyEmail = function verifyEmail(spToken, callback) {
    this.service.verifyEmail(spToken, callback);
  };

  UserStore.prototype.logout = function logout(callback) {
    var self = this;

    this.service.logout(function (err) {
      if (err) {
        return callback(err);
      }

      self.reset();
      self.emitChange();

      callback();
    });
  };

  UserStore.prototype.resolveSession = function resolveSession(callback) {
    var self = this;

    if (_sessionResolved) {
      return callback && callback(_sessionError, _session);
    }

    this.service.me(function (err, result) {
      self.reset();

      _sessionResolved = true;

      if (err) {
        _sessionError = err;
      } else {
        _session = result;
      }

      if (callback) {
        callback(_sessionError, _session);
      }

      self.emitChange();
    });
  };

  UserStore.prototype.reset = function reset(resolved) {
    _session = false;
    _sessionError = null;
    _sessionResolved = false;
  };

  return UserStore;
})(BaseStore);

var userStore = new UserStore();

_app2['default'].on('ready', function (context) {
  userStore.init(context);
  context.getDispatcher().register(function (payload) {
    switch (payload.actionType) {
      case UserConstants.USER_LOGIN:
        userStore.login(payload.options, payload.callback);
        break;
      case UserConstants.USER_LOGOUT:
        userStore.logout(payload.callback);
        break;
      case UserConstants.USER_REGISTER:
        userStore.register(payload.options, payload.callback);
        break;
      case UserConstants.USER_FORGOT_PASSWORD:
        userStore.forgotPassword(payload.options, payload.callback);
        break;
      case UserConstants.USER_CHANGE_PASSWORD:
        userStore.changePassword(payload.options, payload.callback);
        break;
      case UserConstants.USER_VERIFY_EMAIL:
        userStore.verifyEmail(payload.options.spToken, payload.callback);
        break;
    }
    return true;
  });
});

module.exports = userStore;
