// react-stormpath/lib/Context.js
"use strict";

exports.__esModule = true;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var contextInstance = null;

var Context = (function () {
  function Context() {
    _classCallCheck(this, Context);

    if (contextInstance) {
      return contextInstance;
    }

    this.router = null;
    this.dispatcher = null;
    this.endpoints = null;

    contextInstance = this;
  }

  Context.getInstance = function getInstance() {
    if (!contextInstance) {
      return new Context();
    }
    return contextInstance;
  };

  Context.prototype.setRouter = function setRouter(router) {
    this.router = router;
  };

  Context.prototype.getRouter = function getRouter() {
    return this.router;
  };

  Context.prototype.setDispatcher = function setDispatcher(dispatcher) {
    this.dispatcher = dispatcher;
  };

  Context.prototype.getDispatcher = function getDispatcher() {
    return this.dispatcher;
  };

  Context.prototype.setEndpoints = function setEndpoints(endpoints) {
    this.endpoints = endpoints;
  };

  Context.prototype.getEndpoints = function getEndpoints() {
    return this.endpoints || {};
  };

  return Context;
})();

exports["default"] = Context;
module.exports = exports["default"];
