// react-stormpath/es6/components/Authenticated.js
'use strict';

exports.__esModule = true;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _storesUserStore = require('./../stores/UserStore');

var _storesUserStore2 = _interopRequireDefault(_storesUserStore);

var Authenticated = (function (_React$Component) {
  _inherits(Authenticated, _React$Component);

  function Authenticated() {
    _classCallCheck(this, Authenticated);

    _React$Component.call(this);
    this.onChangeListener = null;
    this.state = {
      authenticated: null
    };
  }

  Authenticated.prototype.onChange = function onChange() {
    var self = this;
    _storesUserStore2['default'].isAuthenticated(function (err, authenticated) {
      if (self.onChangeListener !== null) {
        self.setState({ authenticated: authenticated === true });
      }
    });
  };

  Authenticated.prototype.componentWillMount = function componentWillMount() {
    this.onChangeListener = this.onChange.bind(this);
    _storesUserStore2['default'].addChangeListener(this.onChangeListener);
    this.onChange();
  };

  Authenticated.prototype.componentWillUnmount = function componentWillUnmount() {
    _storesUserStore2['default'].removeChangeListener(this.onChangeListener);
    this.onChangeListener = null;
  };

  Authenticated.prototype.render = function render() {
    return this.state.authenticated === true ? this.props.children : null;
  };

  return Authenticated;
})(_react2['default'].Component);

exports['default'] = Authenticated;
module.exports = exports['default'];
