// react-stormpath/es6/components/VerifyEmailView.js
'use strict';

exports.__esModule = true;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _componentsLoginLink = require('../components/LoginLink');

var _componentsLoginLink2 = _interopRequireDefault(_componentsLoginLink);

var _actionsUserActions = require('../actions/UserActions');

var _actionsUserActions2 = _interopRequireDefault(_actionsUserActions);

var VerifyEmailView = (function (_React$Component) {
  _inherits(VerifyEmailView, _React$Component);

  function VerifyEmailView() {
    _classCallCheck(this, VerifyEmailView);

    _React$Component.apply(this, arguments);
    this.state = {
      status: 'VERIFYING'
    };
  }

  VerifyEmailView.prototype.componentDidMount = function componentDidMount() {
    var self = this;
    var spToken = this.props.spToken;

    _actionsUserActions2['default'].verifyEmail(spToken, function (err) {
      if (err) {
        self.setState({
          status: 'ERROR'
        });
      } else {
        self.setState({
          status: 'VERIFIED'
        });
      }
    });
  };

  VerifyEmailView.prototype.render = function render() {
    return _react2['default'].createElement(
      'div',
      { className: 'row' },
      _react2['default'].createElement(
        'div',
        { className: 'col-sm-offset-4 col-xs-12 col-sm-4' },
        ({
          VERIFYING: _react2['default'].createElement(
            'p',
            { className: 'alert alert-warning' },
            'We are verifying your account.'
          ),
          VERIFIED: _react2['default'].createElement(
            'p',
            { className: 'alert alert-success' },
            'Your account has has been verified! ',
            _react2['default'].createElement(
              _componentsLoginLink2['default'],
              null,
              'Login Now.'
            )
          ),
          ERROR: _react2['default'].createElement(
            'div',
            { className: 'alert alert-danger' },
            'This email verification link is not valid.'
          )
        })[this.state.status]
      )
    );
  };

  return VerifyEmailView;
})(_react2['default'].Component);

exports['default'] = VerifyEmailView;
module.exports = exports['default'];
