// react-stormpath/es6/index.js
'use strict';

exports.__esModule = true;
exports.init = init;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _app = require('./app');

var _app2 = _interopRequireDefault(_app);

var _Context2 = require('./Context');

var _Context3 = _interopRequireDefault(_Context2);

exports.Context = _Context3['default'];

var _componentsRouter = require('./components/Router');

var _componentsRouter2 = _interopRequireDefault(_componentsRouter);

exports.Router = _componentsRouter2['default'];

var _actionsUserActions = require('./actions/UserActions');

var _actionsUserActions2 = _interopRequireDefault(_actionsUserActions);

exports.actions = _actionsUserActions2['default'];

var _constantsUserConstants = require('./constants/UserConstants');

var _constantsUserConstants2 = _interopRequireDefault(_constantsUserConstants);

exports.constants = _constantsUserConstants2['default'];

var _componentsLoginRoute = require('./components/LoginRoute');

var _componentsLoginRoute2 = _interopRequireDefault(_componentsLoginRoute);

exports.LoginRoute = _componentsLoginRoute2['default'];

var _componentsLogoutRoute = require('./components/LogoutRoute');

var _componentsLogoutRoute2 = _interopRequireDefault(_componentsLogoutRoute);

exports.LogoutRoute = _componentsLogoutRoute2['default'];

var _componentsAuthenticatedRoute = require('./components/AuthenticatedRoute');

var _componentsAuthenticatedRoute2 = _interopRequireDefault(_componentsAuthenticatedRoute);

exports.AuthenticatedRoute = _componentsAuthenticatedRoute2['default'];

var _componentsAuthenticated = require('./components/Authenticated');

var _componentsAuthenticated2 = _interopRequireDefault(_componentsAuthenticated);

exports.Authenticated = _componentsAuthenticated2['default'];

var _componentsNotAuthenticated = require('./components/NotAuthenticated');

var _componentsNotAuthenticated2 = _interopRequireDefault(_componentsNotAuthenticated);

exports.NotAuthenticated = _componentsNotAuthenticated2['default'];

var _componentsLoginLink = require('./components/LoginLink');

var _componentsLoginLink2 = _interopRequireDefault(_componentsLoginLink);

exports.LoginLink = _componentsLoginLink2['default'];

var _componentsLogoutLink = require('./components/LogoutLink');

var _componentsLogoutLink2 = _interopRequireDefault(_componentsLogoutLink);

exports.LogoutLink = _componentsLogoutLink2['default'];

var _componentsLoginForm = require('./components/LoginForm');

var _componentsLoginForm2 = _interopRequireDefault(_componentsLoginForm);

exports.LoginForm = _componentsLoginForm2['default'];

var _componentsRegistrationForm = require('./components/RegistrationForm');

var _componentsRegistrationForm2 = _interopRequireDefault(_componentsRegistrationForm);

exports.RegistrationForm = _componentsRegistrationForm2['default'];

var _componentsResetPasswordForm = require('./components/ResetPasswordForm');

var _componentsResetPasswordForm2 = _interopRequireDefault(_componentsResetPasswordForm);

exports.ResetPasswordForm = _componentsResetPasswordForm2['default'];

var _componentsVerifyEmailView = require('./components/VerifyEmailView');

var _componentsVerifyEmailView2 = _interopRequireDefault(_componentsVerifyEmailView);

exports.VerifyEmailView = _componentsVerifyEmailView2['default'];

var _componentsUserField = require('./components/UserField');

var _componentsUserField2 = _interopRequireDefault(_componentsUserField);

exports.UserField = _componentsUserField2['default'];

var _componentsUserComponent = require('./components/UserComponent');

var _componentsUserComponent2 = _interopRequireDefault(_componentsUserComponent);

exports.UserComponent = _componentsUserComponent2['default'];

// When not using ES6, enable people to use ReactStormpath.init()
// instead of ReactStormpath.default.init().

function init() {
  _app2['default'].init.apply(_app2['default'], arguments);
}

;

exports['default'] = _app2['default'];
